src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -62,8 +63,8 @@ private synchronized void loadExecutions() {
                 }
             }
             if (execution != null) {
-                // TODO JENKINS-37154 we would rather not block here
-                for (StepExecution se : execution.getCurrentExecutions(true).get()) {
+                // JENKINS-37154 sometimes we must block here in order to get accurate results
+                for (StepExecution se : execution.getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
> execution time for one (non-asynchronous) step, right?

Yes, for example.

> if that's taking more than a few seconds, it represents a bit of an issue (should be converted to an async step)

Yes, it should, but I would rather not cause other stuff to break just as a result of poor performance.

> there are a limited […] number of request handler threads, and this completely locks one while waiting

Which is why I would like to avoid blocking at all here. Without some redesigns that does not seem to be possible given an arbitrary caller, but at least the known problem of a deadlock coming from `stop` is solved by this PR, and other _unknown_ problems are at least ameliorated by adding some timeout.